src/LinuxCaptureService/MemoryWatchdog.cpp

```diff
@@ -0,0 +1,112 @@
+// Copyright (c) 2021 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "MemoryWatchdog.h"
+
+#include <absl/strings/match.h>
+#include <absl/strings/numbers.h>
+#include <absl/strings/str_split.h>
+#include <unistd.h>
+
+#include "OrbitBase/Logging.h"
+#include "OrbitBase/ReadFileToString.h"
+
+namespace orbit_linux_capture_service {
+
+std::optional<uint64_t> ExtractMemTotalInKbFromProcMeminfo(std::string_view proc_meminfo) {
+  std::string line;
+  for (size_t i = 0; i <= proc_meminfo.size(); ++i) {
+    if (i == proc_meminfo.size() || proc_meminfo[i] == '\n') {
+      if (!absl::StartsWith(line, "MemTotal:")) {
+        line.clear();
+        continue;
+      }
+
+      std::vector<std::string_view> splits = absl::StrSplit(line, ' ', absl::SkipWhitespace{});
+      if (splits.size() < 3 || splits[2] != "kB") {
+        ERROR("Extracting MemTotal from \"%s\"", line);
+        return std::nullopt;
+      }
+
+      uint64_t mem_total_kb;
+      if (!absl::SimpleAtoi(splits[1], &mem_total_kb)) {
+        ERROR("Parsing MemTotal \"%s\"", splits[1]);
+        return std::nullopt;
+      }
+
+      return mem_total_kb;
+    }
+
+    line.push_back(proc_meminfo[i]);
+  }
+
+  ERROR("Could not find MemTotal in file");
+  return std::nullopt;
+}
+
+std::optional<uint64_t> ReadMemTotalInBytesFromProcMeminfo() {
```

Comment:
It feels a bit weird to do all the file parsing. I'm not 100% sure if this is the same thing and you might have considered this but wold this yield the same result?   

```
long pages = sysconf(_SC_PHYS_PAGES);
long page_size = sysconf(_SC_PAGE_SIZE);
return pages * page_size;
```

Reply:
Hmm, I haven't found anything for getting the RSS of a process. So I guess just doing the parsing is fine. Feel free to disregard.

Reply:
Ah, I didn't know about `_SC_PHYS_PAGES`. From what I can see that gives the same result, at least on our platforms. So that simplifies things...

Reply:
I don't think there's anything for the RSS, after all that's per-process.